plugins/i3s/io/EsriReader.hpp

```diff
@@ -0,0 +1,165 @@
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#pragma once
+#define ARBITER_ZLIB
+#include "../lepcc/src/include/lepcc_c_api.h"
+#include "../lepcc/src/include/lepcc_types.h"
+#include "EsriReceiver.hpp"
+#include "pool.hpp"
+#include <pdal/PointView.hpp>
+#include <pdal/Reader.hpp>
+#include <pdal/util/IStream.hpp>
+#include <pdal/PointLayout.hpp>
+#include <pdal/StageFactory.hpp>
+#include <pdal/SpatialReference.hpp>
+#include <json/json.h>
+#include <arbiter/arbiter.hpp>
+
+#include <array>
+#include <functional>
+#include <queue>
+#include <vector>
+#include <algorithm>
+#include <chrono>
+#include <Eigen/Geometry>
+#include <gdal.h>
+#include <ogr_spatialref.h>
+
+namespace pdal
+{
+
+class EsriReader : public Reader
+{
+public:
+    BOX3D createBounds();
+
+protected:
+    virtual void initInfo() = 0;
+    virtual void buildNodeList(std::vector<int>& nodes, int pageIndex) = 0;
+    virtual std::vector<char> fetchBinary(std::string url, std::string attNum,
+            std::string ext) const = 0;
+
+
+
+    std::unique_ptr<ILeStream> m_stream;
+    std::unique_ptr<arbiter::Arbiter> m_arbiter;
+
+    I3SArgs m_args;
+    Json::Value m_info;
+    std::mutex m_mutex;
+    BOX3D m_bounds;
+    int m_nodeCap;
+
+    gzip::Decompressor m_decomp;
+
+    //Spatial Reference variables
+    SpatialReference m_nativeSrs;
+    SpatialReference m_ecefSrs;
+
+    typedef void* ReferencePtr;
+    typedef void* TransformPtr;
+    ReferencePtr m_nativeRef;
+    ReferencePtr m_ecefRef;
+
+    TransformPtr m_toEcefTransform;
+    TransformPtr m_toNativeTransform;
+
+    struct dimData
+    {
+        int key;
+        std::string dataType;
+    };
+    std::map<Dimension::Id, dimData> m_dimMap;
+    std::map<std::string, Dimension::Id> m_dimensions {
```

Comment:
I don't think `m_dimensions` needs to be a member, does it?  I believe it could be a `const` in the unnamed namespace in the CPP file.